heat/cluster/kmedians.py

```diff
@@ -0,0 +1,118 @@
+import heat as ht
+from heat.cluster._kcluster import _KCluster
+
+
+class KMedians(_KCluster):
+    def __init__(self, n_clusters=8, init="random", max_iter=300, tol=1e-4, random_state=None):
+        """
+        K-Medians clustering algorithm. An implementation of Lloyd's algorithm [1].
+        Uses the Manhattan (City-block, :math:`L_1`) metric for distance calculations
+
+        Parameters
+        ----------
+        n_clusters : int, optional, default: 8
+            The number of clusters to form as well as the number of centroids to generate.
+        init : str or DNDarray
+            Method for initialization, defaults to ‘random’:
+                - ‘k-medians++’ : selects initial cluster centers for the clustering in a smart way to speed up convergence [2].
+                - ‘random’: choose k observations (rows) at random from data for the initial centroids. \n
+                - ``DNDarray``: gives the initial centers, should be of Shape = (n_clusters, n_features)
+        max_iter : int, default: 300
+            Maximum number of iterations of the k-means algorithm for a single run.
+        tol : float, default: 1e-4
+            Relative tolerance with regards to inertia to declare convergence.
+        random_state : int
+            Determines random number generation for centroid initialization.
```

Comment:
Could we have a reference here?

Reply:
For what exactly?

Reply:
K-Medians paper

Reply:
Oded Kariv and S Louis Hakimi.  An algorithmic approach to network location problems. ii:  Thep-medians.SIAM Journal on Applied Mathematics, 37(3):539–560, 1979.